backend/src/main/kotlin/hu/bme/sch/cmsch/model/UserDetailsByInternalIdMappingEntity.kt

```diff
@@ -0,0 +1,100 @@
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.fasterxml.jackson.annotation.JsonView
+import hu.bme.sch.cmsch.admin.*
+import hu.bme.sch.cmsch.dto.Edit
+import hu.bme.sch.cmsch.dto.FullDetails
+import hu.bme.sch.cmsch.dto.Preview
+import hu.bme.sch.cmsch.model.*
+import javax.persistence.*
+
+@Entity
+@Table(name="userDetailsByInternalId")
+data class UserDetailsByInternalIdMappingEntity(
+    @Id
+    @GeneratedValue
+    @JsonView(value = [ Edit::class ])
+    @property:GenerateInput(type = INPUT_TYPE_HIDDEN, visible = true, ignore = true)
+    @property:GenerateOverview(visible = false)
+    override var id: Int = 0,
+
+    @JsonView(value = [ Edit::class ])
+    @Column(nullable = false)
+    @property:GenerateInput(order = 1, label = "PéK internal id",
+        note = "Ez módosítható eseti hiba kezelésre", enabled = true)
+    @property:GenerateOverview(visible = false)
+    @property:ImportFormat(ignore = false, columnId = 0)
+    var internalId: String? = null,
+
+    @JsonView(value = [ Edit::class, FullDetails::class ])
+    @property:GenerateInput(order = 2, label = "Neptun kód", enabled = true,
+        note = "Ez módosítható eseti hiba kezelésre", maxLength = 6)
+    @property:GenerateOverview(columnName = "Neptun", order = 2)
+    @property:ImportFormat(ignore = false, columnId = 1)
+    var neptun: String? = null,
+
+    @JsonView(value = [ Edit::class, FullDetails::class ])
+    @Enumerated(EnumType.STRING)
+    @property:GenerateInput(type = INPUT_TYPE_BLOCK_SELECT, order = 7, label = "Jogkör",
+        source = [ "GUEST", "BASIC", "ATTENDEE", "PRIVILEGED", "STAFF", "ADMIN", "SUPERUSER" ], minimumRole = RoleType.ADMIN,
+        note = "BASIC = belépett, STAFF = rendező, ADMIN = minden jog")
+    @property:GenerateOverview(visible = false)
+    @property:ImportFormat(ignore = false, columnId = 6, enumSource = RoleType::class, defaultValue = "GUEST")
```

Comment:
Column ids must start with 0 (which is true in this case), and must be unique incremental with no gaps. 0,1,2,3,..,n
Used for CSV import and export.

Reply:
Alright, this columnId annotation is not added to the Id and the columnName attribute, my guess is that it is not needed at the columnName because it is an entity, should I add it to the Id attribute? Will we export and import these Id-s, or because it is autogenerated we don't have to?

Reply:
_See my other comments._